overlord/state/task.go

```diff
@@ -531,6 +531,20 @@ func (ts *TaskSet) AddAll(anotherTs *TaskSet) {
 	}
 }
 
+// AddAllWithEdges adds all the tasks in the argument set to the target
+// set ts and also adds all TaskSetEddges. Duplicated TaskSetEdges are
+// an error.
+func (ts *TaskSet) AddAllWithEdges(anotherTs *TaskSet) error {
+	ts.AddAll(anotherTs)
```

Comment:
Probably not an issue, but we're mutating ts here while the loop below can still fail. So we must be careful not to use the updated tasket if AddAllWithEdges reported an error.

Reply:
Good point, still scratching my head about what the best way to avoid this is - its an internal erorr so hopefully not a big deal.